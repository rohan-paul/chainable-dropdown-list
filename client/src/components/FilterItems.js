import React, { Component } from 'react';

class FilterItems extends React.Component {
  render () {
    return (
      <div className="filter-items">
        {this.props.data.map(item => {
          return (<div key={item.name} className="filter-item">{item.name}</div>)
        }
      )}
      </div>

    );
  }
};

export default FilterItems;