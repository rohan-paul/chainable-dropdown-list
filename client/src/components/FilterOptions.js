import React, { Component } from 'react';

class FilterOptions extends React.Component {

  changeOption = (type, e) => {
    let val = e.target.value;
    this.props.changeOptionProp(val, type);
  }

  render () {
    return (
      <div className="filter-options">
        <div className="filter-option">

          <label>Car Make</label>
          <select id="carMake" placeholder="Search by tags" value={this.props.carMake} onChange={this.changeOption.bind(this, 'carMake')}>

          {this.props.carMakeOptions.map(option => (<option key={option} value={option}>{option}</option> ))}
          </select>

          <label>Model</label>
          <select id="model" value={this.props.model} onChange={this.changeOption.bind(this, 'model')}>

          {this.props.modelOptions.map(option => (<option key={option} value={option}>{option}</option> ))}
          </select>

        </div>
      </div>
    );
  }
};

export default FilterOptions;