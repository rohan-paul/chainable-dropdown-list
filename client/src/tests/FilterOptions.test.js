import React from 'react';
// import ReactDOM from 'react-dom';
import { ReactTestUtils } from 'react-dom/test-utils'; // ES6
import expect from 'expect';
import toJson from 'enzyme-to-json';
import FilterOptions from '../components/FilterOptions';
import { shallow, mount } from 'enzyme';


describe('FilterOptions', () => {
  it('should exist', () => {
    expect(FilterOptions).toBeTruthy();
  });
});

describe('FilterOptions', function () {
  it('loads without problems', function () {
    require('../components/FilterOptions');
  });
});

it('shallow render without crashing', () => {
  const wrapper = shallow(<FilterOptions
      data={ [ 1, 2, 3] }
      carMakeOptions={ [ 1, 2, 3] }
      modelOptions={ [ 1, 2, 3] }
      changeOption={ [ 1, 2, 3] }
   />);
  expect(wrapper).toMatchSnapshot();
});

/* describe('FilterOptions', function () {

  it('changes without problems', function () {
    var FilterOptions =  ReactTestUtils.renderIntoDocument(<FilterOptions/>);

    const inputNode = ReactDOM.findDOMNode(FilterOptions.refs.input);

    const newValue = 'some text';
    inputNode.value = newValue;
    ReactTestUtils.Simulate.change(inputNode);

    const nameNode = ReactDOM.findDOMNode(FilterOptions.refs.name);
    expect(nameNode.textContent).toEqual(newValue);
  });
}); */