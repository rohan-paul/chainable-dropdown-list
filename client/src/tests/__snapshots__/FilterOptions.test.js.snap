// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterOptions
    carMakeOptions={
        Array [
            1,
            2,
            3,
          ]
    }
    changeOption={
        Array [
            1,
            2,
            3,
          ]
    }
    data={
        Array [
            1,
            2,
            3,
          ]
    }
    modelOptions={
        Array [
            1,
            2,
            3,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="filter-option"
>
        <label>
                Car Make
        </label>
        <select
                id="carMake"
                onChange={[Function]}
                placeholder="Search by tags"
                value={undefined}
        >
                <option
                        value={1}
                >
                        1
                </option>
                <option
                        value={2}
                >
                        2
                </option>
                <option
                        value={3}
                >
                        3
                </option>
        </select>
        <label>
                Model
        </label>
        <select
                id="model"
                onChange={[Function]}
                value={undefined}
        >
                <option
                        value={1}
                >
                        1
                </option>
                <option
                        value={2}
                >
                        2
                </option>
                <option
                        value={3}
                >
                        3
                </option>
        </select>
</div>,
      "className": "filter-options",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Car Make
</label>,
          <select
            id="carMake"
            onChange={[Function]}
            placeholder="Search by tags"
            value={undefined}
>
            <option
                        value={1}
            >
                        1
            </option>
            <option
                        value={2}
            >
                        2
            </option>
            <option
                        value={3}
            >
                        3
            </option>
</select>,
          <label>
            Model
</label>,
          <select
            id="model"
            onChange={[Function]}
            value={undefined}
>
            <option
                        value={1}
            >
                        1
            </option>
            <option
                        value={2}
            >
                        2
            </option>
            <option
                        value={3}
            >
                        3
            </option>
</select>,
        ],
        "className": "filter-option",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Car Make",
          },
          "ref": null,
          "rendered": "Car Make",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value={1}
>
                1
</option>,
              <option
                value={2}
>
                2
</option>,
              <option
                value={3}
>
                3
</option>,
            ],
            "id": "carMake",
            "onChange": [Function],
            "placeholder": "Search by tags",
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "value": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "value": 2,
              },
              "ref": null,
              "rendered": 2,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": 3,
                "value": 3,
              },
              "ref": null,
              "rendered": 3,
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Model",
          },
          "ref": null,
          "rendered": "Model",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value={1}
>
                1
</option>,
              <option
                value={2}
>
                2
</option>,
              <option
                value={3}
>
                3
</option>,
            ],
            "id": "model",
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "value": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "value": 2,
              },
              "ref": null,
              "rendered": 2,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": 3,
                "value": 3,
              },
              "ref": null,
              "rendered": 3,
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="filter-option"
>
          <label>
                    Car Make
          </label>
          <select
                    id="carMake"
                    onChange={[Function]}
                    placeholder="Search by tags"
                    value={undefined}
          >
                    <option
                              value={1}
                    >
                              1
                    </option>
                    <option
                              value={2}
                    >
                              2
                    </option>
                    <option
                              value={3}
                    >
                              3
                    </option>
          </select>
          <label>
                    Model
          </label>
          <select
                    id="model"
                    onChange={[Function]}
                    value={undefined}
          >
                    <option
                              value={1}
                    >
                              1
                    </option>
                    <option
                              value={2}
                    >
                              2
                    </option>
                    <option
                              value={3}
                    >
                              3
                    </option>
          </select>
</div>,
        "className": "filter-options",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Car Make
</label>,
            <select
              id="carMake"
              onChange={[Function]}
              placeholder="Search by tags"
              value={undefined}
>
              <option
                            value={1}
              >
                            1
              </option>
              <option
                            value={2}
              >
                            2
              </option>
              <option
                            value={3}
              >
                            3
              </option>
</select>,
            <label>
              Model
</label>,
            <select
              id="model"
              onChange={[Function]}
              value={undefined}
>
              <option
                            value={1}
              >
                            1
              </option>
              <option
                            value={2}
              >
                            2
              </option>
              <option
                            value={3}
              >
                            3
              </option>
</select>,
          ],
          "className": "filter-option",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Car Make",
            },
            "ref": null,
            "rendered": "Car Make",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value={1}
>
                  1
</option>,
                <option
                  value={2}
>
                  2
</option>,
                <option
                  value={3}
>
                  3
</option>,
              ],
              "id": "carMake",
              "onChange": [Function],
              "placeholder": "Search by tags",
              "value": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "value": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": 2,
                  "value": 2,
                },
                "ref": null,
                "rendered": 2,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": 3,
                  "value": 3,
                },
                "ref": null,
                "rendered": 3,
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Model",
            },
            "ref": null,
            "rendered": "Model",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value={1}
>
                  1
</option>,
                <option
                  value={2}
>
                  2
</option>,
                <option
                  value={3}
>
                  3
</option>,
              ],
              "id": "model",
              "onChange": [Function],
              "value": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "value": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": 2,
                  "value": 2,
                },
                "ref": null,
                "rendered": 2,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": 3,
                  "value": 3,
                },
                "ref": null,
                "rendered": 3,
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
